### DISCUSSION TITLE
almost there, but problems with one device on the compatibility list.

### DISCUSSION THREAD

**Discussion by tanderson99999** [bemasher/rtlamr#177] [Q&A]
Using "rtlamr -msgtype=scm -format=json" i was able to get the output from my 100GDLAN gas meter no problem.   So i know my "NooElec NESDR Nano 2+" is setup properly and your software compiled properly.

I am having problems getting data from my itron 50w water meter.  

i am trying to use
rtlamr -msgtype=all -format=json

This is the meter.  the small print says "ERT Model:  50W ERW-1008-501,502,503,504,505"
![20210121_075700](https://user-images.githubusercontent.com/79128801/108020369-7a936b00-6fd9-11eb-8a6a-326ead411083.jpg)

i believe it should use scm and have a type of 03 with an ID of 21537736, but i never see a device with that ID or a type other then 12 ( gas meters).  I have also looked and no devices match the usage.  My SDR is in the same room 3 feet away from the meter.

Is there sometime extra i need to do for the Itron 50W?

This is the FCC link for this Itron model https://fccid.io/EO940GN-1  
looks like it runs at 910-920 MHz

**Comment by tanderson99999** [comment]
I tried the following, still no water meters showing up, is a difference of 2mhz enough to find the signal 

	rtlamr -centerfreq=902000000 -msgtype=all -format=json
	rtlamr -centerfreq=904000000 -msgtype=all -format=json
	rtlamr -centerfreq=906000000 -msgtype=all -format=json
	rtlamr -centerfreq=908000000 -msgtype=all -format=json
	rtlamr -centerfreq=910000000 -msgtype=all -format=json
	rtlamr -centerfreq=912000000 -msgtype=all -format=json
	rtlamr -centerfreq=914000000 -msgtype=all -format=json
	rtlamr -centerfreq=916000000 -msgtype=all -format=json
	rtlamr -centerfreq=918000000 -msgtype=all -format=json
	rtlamr -centerfreq=920000000 -msgtype=all -format=json  
	rtlamr -centerfreq=922000000 -msgtype=all -format=json

**Reply by bemasher** [reply]
Based on the label alone, it should be ERT compatible and receivable with rtlamr.

**Comment by Mudary** [comment]
My water is a Neptune and uses -msgtype=r900

**Reply by tanderson99999** [reply]
im under the impression that -msgtype=all includes r900 is that not the case?

**Reply by bemasher** [reply]
It does: https://github.com/bemasher/rtlamr/blob/master/main.go#L61-L68

**Reply by tanderson99999** [reply]
If you don't mind me asking how often does your water meter transmit?  I assume water needs to be running to generate power for it to transmit?  or is there an internal battery that charges when there is flow and then continues to transmit on a schedule?

**Reply by bemasher** [reply]
Pretty much all ERT and R900-based transceivers are battery operated. ERT meters can operate in bubble-up mode or wake-up mode, meaning they transmit at fixed intervals or transmit only when woken by an appropriate signal.

**Comment by Nils154** [comment]
My Neptune only reports if I explicitly include R900 as a type (along with SCM for the gasmeter). No need for any other setting changes from the default.

--
Nils Gokemeijer (Sent from iPhone)

**Reply by tanderson99999** [reply]
weird, but worth trying, i will try each center freq again this time with r900

**Reply by tanderson99999** [reply]
If you don't mind me asking how often does your water meter transmit?  I assume water needs to be running to generate power for it to transmit?  or is there an internal battery that charges when there is flow and then continues to transmit on a schedule?

**Comment by Nils154** [comment]
My meter says Neptune E-coder R900i on the front, it is in the basement, only a few years old.
It transmits about every 3 minutes.

I worry that you need enough cpu resources to decode the signal, but it works on my RPI2

--
Nils Gokemeijer (Sent from iPhone)

**Reply by tanderson99999** [reply]
Thanks for the reply, i ordered another usb rtl device to use on my new PC to see if that has more luck.  This will also let me play around with settings without having to disconnect the currently working setup i have for the gas meter.

**Comment by tanderson99999** [comment]
I have changed strategy a bit, i bought another RTL device (same model) so i can work on a different PC while leaving the other unit alone.  The new computer is much faster (Ryzen 9 5950x) and picks up more then the old one.  I used to have ll- and ll+ messages appearing on the Rtl_tcp.exe console but those no longer happen on the new setup.  I am now seeing a small number of SCM+ devices that never showed up before.  I believe this is because my CPU is now fast enough to process them.  Using the -msgtype=all on the slower PC these are not visible, however if i use -msgtype=scm+ they do show up, i think this is what Nils154 was referring to where =all does not always work.

Now that i have confirmed that the 'all' option works on this PC i started my way thought the following list, running on each frequency for at least 1 hour, documenting the start and end times and the ID's that were found.  I now have 5 additional ID

```
rtlamr -centerfreq=900000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=901000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=902000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=903000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=904000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=905000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=906000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=907000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=908000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=909000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=910000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=911000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=912000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=913000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=914000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=915000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=916000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=917000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=918000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=919000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=920000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=921000000 -msgtype=all -format=json		Start Time:			End Time:		Found:		
rtlamr -centerfreq=922000000 -msgtype=all -format=json		Start Time:			End Time:		Found:
```

These are the new events i see

```json
{"Time":"2021-03-01T15:36:41.1318664-07:00","Offset":0,"Length":0,"Type":"SCM","Message":{"ID":24309235,"Type":4,"TamperPhy":1,"TamperEnc":2,"Consumption":15463848,"ChecksumVal":32853}}

{"Time":"2021-03-02T02:06:15.6181301-07:00","Offset":0,"Length":0,"Type":"SCM","Message":{"ID":1354714,"Type":1,"TamperPhy":0,"TamperEnc":1,"Consumption":11261610,"ChecksumVal":57365}}

{"Time":"2021-03-02T15:07:29.6579203-07:00","Offset":0,"Length":0,"Type":"SCM+","Message":{"FrameSync":5795,"ProtocolID":30,"EndpointType":156,"EndpointID":92341707,"Consumption":929098,"Tamper":768,"PacketCRC":56996}}

{"Time":"2021-03-02T16:28:36.5580566-07:00","Offset":0,"Length":0,"Type":"SCM+","Message":{"FrameSync":5795,"ProtocolID":30,"EndpointType":156,"EndpointID":91120160,"Consumption":719136,"Tamper":512,"PacketCRC":40036}}

{"Time":"2021-03-02T18:52:33.6130526-07:00","Offset":0,"Length":0,"Type":"SCM+","Message":{"FrameSync":5795,"ProtocolID":30,"EndpointType":156,"EndpointID":92222874,"Consumption":833940,"Tamper":256,"PacketCRC":64906}}
```

Looking at my meter photos posted above i can see that it was made in 2003, from the itron documents i have read SCM+ only came out in late 2013 so its doubtful that those would be mine, however it looks like new meters issued in my city are showing up with SCM+

I am now trying some experiments to see if the unit with ID 24309235 and type 4 is my meter by running the water at specific times of day and looking for spikes in the numbers.  So far it doesn't look like that meter is mine.

bemasher wrote in his note 6 days ago, "meaning they transmit at fixed intervals or transmit only when woken by an appropriate signal."

I'm am interested in the statement "transmit only when woken by an appropriate signal".  Do we know anything about that signal, is it always the same, does RTLamr support it? is a challenge and response? what frequency is it on?  My though is that if my meter uses that signal then i could wait a month for the meter reading truck to drive by and look at the data to see if i could find that message and replay it.

**Reply by bemasher** [reply]
Rtl-sdr's are receive only. There are a lot of reasons you wouldn't want to wake your meter up anyway: https://github.com/bemasher/rtlamr/issues/103#issuecomment-425316891

If you haven't already seen a message with a serial number matching the label of your meter, you probably aren't going to unless you are listening when it is woken up.
